src/fees/FeeVerifier.sol

```diff
@@ -5,32 +5,28 @@ import {Ownable} from 'openzeppelin-contracts/contracts/access/Ownable.sol';
 import {IParam} from '../interfaces/IParam.sol';
 import {IFeeCalculator} from '../interfaces/IFeeCalculator.sol';
 
-abstract contract FeeVerifier is Ownable {
+abstract contract FeeGenerator is Ownable {
     error LengthMismatch();
 
     event FeeCalculatorSet(bytes4 indexed selector, address indexed to, address indexed feeCalculator);
 
-    address internal constant _NATIVE = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
     address internal constant _DUMMY_TO_ADDRESS = address(0);
     bytes4 internal constant _NATIVE_FEE_SELECTOR = 0xeeeeeeee;
 
     mapping(bytes4 selector => mapping(address to => address feeCalculator)) public feeCalculators;
 
-    /// @notice Get logics, msg.value and fees that contains fee
-    function getLogicsAndFees(
+    /// @notice Get logics and msg.value that contains fee
+    function getUpdatedLogicsAndMsgValue(
         IParam.Logic[] memory logics,
         uint256 msgValue
-    ) external view returns (IParam.Logic[] memory, uint256, IParam.Fee[] memory) {
+    ) external view returns (IParam.Logic[] memory, uint256) {
         // Update logics
         logics = getLogicsDataWithFee(logics);
 
         // Update value
         msgValue = getMsgValueWithFee(msgValue);
 
-        // Get fees
-        IParam.Fee[] memory fees = getFeesByLogics(logics, msgValue);
```

Comment:
In my opinion, we could still keep the fee as return value because it could show how many fee we will charge in router, even if the fee is useless when executed.

Reply:
In agentImplementation, we'll emit events of all fees. So we could see all the fee through event. Prefer remove fees here.